core/src/main/java/com/linecorp/armeria/common/MediaTypeSet.java

```diff
@@ -150,15 +150,35 @@ public int size() {
         return match(ranges);
     }
 
+    /**
+     * Finds the {@link MediaType} in this {@link List} that matches the specified media range.
+     *
+     * @return the {@link MediaType} that matches the specified media range.
+     *         {@link Optional#empty()} if there are no matches
+     */
+    public Optional<MediaType> match(MediaType range) {
+        requireNonNull(range, "range");
+        for (MediaType candidate : mediaTypes) {
+            if (candidate.belongsTo(range)) {
+                // With only one specified range, there is no way for candidates to have priority over each
+                // other, we just return the first match.
+                return Optional.of(candidate);
+            }
+        }
+        return Optional.empty();
+    }
+
     /**
      * Finds the {@link MediaType} in this {@link List} that matches one of the specified media ranges.
      *
      * @return the most preferred {@link MediaType} that matches one of the specified media ranges.
      *         {@link Optional#empty()} if there are no matches or {@code ranges} does not contain
```

Comment:
Don't we need to update this doc as well because there's no `ranges` anymore?

Reply:
@anuraaga Could you update the Javadoc?